### ISSUE TITLE
Allows users to specify subplot dimensions manually

### ISSUE THREAD

**Issue by seisman** [GenericMappingTools/gmt#1344]
Currently, subplot dimensions are always automatically determined by subplot size (`-F`) and `-C`.

It would be better if users can specify the subplot dimensions.  Here is a proposal: 

For commands in subplots:

1. If no `-J` is given, then the subplot dimensions are automatically determined
2. If `-J` is given with a `?` (maybe `-JX?/?` for linear projection), we can replace `?` with the auto-calculated value
3. If `-J` is given without `?`, we may assume users give the full `-J` option and don't append any values.

**Comment by PaulWessel** [comment]
I don't follow, you will need to expand on this. Looking at the three cases:

1. If no **-J** then we default to either -JQ15c or -JX15c.  I guess this gets auto-scaled to fit the size specified.  This is a bit unclear to me.  So you would want -R -J to work together to set -Fs<size>?
2. This is what we do I think?
3. Likewise, this is the expected behavior, no?

**Comment by seisman** [comment]
Using gmt 6.0.0_1213772_2019.08.08.
```
gmt begin map
gmt subplot begin 2x2 -Fs5c/5c -A
    gmt basemap -R0/10/0/10 -Baf -c
    gmt basemap -R0/10/0/10 -JX? -Baf -c
    gmt basemap -R0/10/0/10 -JM? -Baf -c
    gmt basemap -R0/10/0/10 -JM2c -Baf -c
gmt subplot end
gmt end show
```

This is what I get.

![image](https://user-images.githubusercontent.com/3974108/62752376-4084f880-ba35-11e9-9e67-e0cde589c44b.png)

Panels a, b, c are all good. Panel d is expected to be 2 cm wide, but changing the `-JM2c` doesn't change the subplot size.

**Comment by PaulWessel** [comment]
OK, I agree that does not look right - will see what is going on.  I think I know.

**Comment by PaulWessel** [comment]
Found how to fix this, so not a problem.  However, my fix then broke ex06.sh.  The auto-stuff is not clear for -JXa/b settings and probably not fully tested for logarithmic and power scale projections.  I will try to do more tonight.  The basic fix is in commit cartautoscaling.

**Comment by seisman** [comment]
I'm pretty confused about exmod/ex06.sh. In this script, the subplots rose and histogram use `-JX3.6i` and `-JX4.8i/2.4i`, respectively. It seems I can change the rose size but cannot change the histogram size.

**Comment by PaulWessel** [comment]
Yes, so unlike maps, Cartesian plots can have two scales. Our docs says:

> Any **-J** option passed when plotting subplots must give ? as scale or width since the dimensions of the map are completely determined by the subplot size and your region. Specifying map width will result in an error. For Cartesian plots: If you want the scale to apply equally to both dimensions then you must specify **-Jx** [The default **-JX** will fill the subplot by using unequal scales]

So perhaps ex06 should really do -JX? to get unequal scaling.  Also, if you do specify a fixed size, then there is no centering etc.  The goal of subplot was to make things simple.  If people want to do strange things and demand to use -X and -Y to move around then that is a problem (for them).  I know you were just testing with -JM2c above but it seems to me that fixed size is not a good model for subplots - they should autoscale.  You can affect this not by changing the size but by shrinking the canvas via -C.

**Comment by seisman** [comment]
I'm OK with the autoscale model. 

> Any -J option passed when plotting subplots must give ? as scale or width since the dimensions of the map are completely determined by the subplot size and your region. Specifying map width will result in an error.

This is not true now. Specifying map width doesn't raise an error.

**Comment by seisman** [comment]
I think this is fixed in #1354. Closed.

**Comment by PaulWessel** [comment]
ex06.sh fixed per #1359.
